### PR TITLE
Referrals: deep link for claiming guest pass

### PR DESCRIPTION
**Pull request by SergioEstevao**
| 📘 Part of: #2083  |  
|:---:|

Fixes # 

Update the referrals code to support deep links and proper check your current subscription status to see if it should show the claim screen or show you `The offer is not available screen`

At the moment we are still not validating the redeem code on the server so all `/redeem/*` urls will be valid.

| 1 | 2 | 3 |
| - | - | - |
| Simulator Screenshot - iPhone 15 Pro - 2024-09-20 at 19 06 52 |  Simulator Screenshot - iPhone 15 Pro - 2024-09-20 at 19 06 25 |  Simulator Screenshot - iPhone 15 Pro - 2024-09-20 at 19 06 23 |

## To test

1. Start the app
2. Ensure that you have the referrals FF enabled
3. Switch the app to the background
4. Open a redeem deep link 
    - **If testing on the simulator**: Open the the terminal and write this: `xcrun simctl openurl booted 'http://pocketcasts.com/redeem/test'` . You need to have only one simulator open 
    - If testing on the device: Send an email to yourself with the following link in the content: `http://pocketcasts.com/redeem/test`.  Open the email on the device and tap on the link
5. Check that  the app opens one of the follow:
   - If you don't have a subscription you see the Claim Guest pass screen 
   - If you already have an existing subscription you should see the `You already have a subscription screen`
6. Remember test both scenarios on Settings -> Developer you can change your subscription status 
7. If you saw the Claim guest pass screen, if you tap on `Not now` you should screen a cell on the profile allowing to go back to the claim screen.

## Checklist

- [x] I have considered if this change warrants user-facing release notes and have added them to `CHANGELOG.md` if necessary.
- [x] I have considered adding unit tests for my changes.
- [x] I have updated (or requested that someone edit) [the spreadsheet](https://docs.google.com/spreadsheets/d/107jqrutZhU0fVZJ19SBqxxVKbV2NWSdQC9MFYdLiAxc/edit?usp=sharing) to reflect any new or changed analytics.

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">This PR is still a Draft: some checks will be skipped.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: podcasts/podcasts.entitlements [hunk 1]
```diff
@@ -17,7 +17,7 @@
 		<string>webcredentials:play.pocketcasts.com</string>
 		<string>applinks:pcast.pocketcasts.net</string>
 		<string>applinks:pca.st</string>
-		<string>applinks:pocketcasts.com</string>
+		<string>applinks:pocketcasts.com?mode=developer</string>
```
Comment (SergioEstevao) [inline podcasts/podcasts.entitlements]: Temporary change to allow us to test in real time changes in the `apple-app-site-association` file

File: podcasts/podcasts.prototype.entitlements [hunk 2]
```diff
@@ -11,7 +11,7 @@
 		<string>webcredentials:play.pocketcasts.com</string>
 		<string>applinks:pcast.pocketcasts.net</string>
 		<string>applinks:pca.st</string>
-		<string>applinks:pocketcasts.com</string>
+		<string>applinks:pocketcasts.com?mode=developer</string>
```
Comment (SergioEstevao) [inline podcasts/podcasts.prototype.entitlements]: Temporary change to allow us to test in real time changes in the apple-app-site-association file

File: podcasts/podcastsDebug.entitlements [hunk 3]
```diff
@@ -15,7 +15,7 @@
 		<string>webcredentials:play.pocketcasts.com</string>
 		<string>applinks:pcast.pocketcasts.net</string>
 		<string>applinks:pca.st</string>
-		<string>applinks:pocketcasts.com</string>
+		<string>applinks:pocketcasts.com?mode=developer</string>
```
Comment (SergioEstevao) [inline podcasts/podcastsDebug.entitlements:18]: Temporary change to allow us to test in real time changes in the apple-app-site-association file

File: podcasts/podcasts.entitlements (hunk 1 above)
Comment (bjtitus) [reply podcasts/podcasts.entitlements]: Should this be included in `trunk` or only in PRs for testing?

**Review by bjtitus — APPROVED** [review]
✅ Verified the unavailable offer for Plus users
✅ Verified the unavailable offer for Non-Plus users
✅ Verified the Guest Pass cell in Profile after dismissing

The only question I have is about including the `?mode=developer` in `trunk` or if we'd be better off excluding that. I guess my main concern would be getting caught in automated review for TestFlight builds.

:shipit:

File: podcasts/podcasts.entitlements (hunk 1 above)
Comment (SergioEstevao) [reply podcasts/podcasts.entitlements]: I will revert it now. So this doesn't go in by mistake later on!

Comment (SergioEstevao) [reply podcasts/podcasts.entitlements]: Done here: https://github.com/Automattic/pocket-casts-ios/pull/2178/commits/8001eec8a77a9d843663793faaa895566b649f36

**Pull request merged**